MyProject.GeneratedMSBuildEditorConfig.editorconfig' could not be found. Receiving error MSB6006: "csc.dll" exited with code 139

**Version Used**:  7.0.302

**Steps to Reproduce**:

I have an ASP.NET Identity app in GitHub actions, and when building the web app (with code analysts enabled) I'm receiving the following error:

`MSB6006: "csc.dll" exited with code 139.`

Following the instructions in [this comment](https://github.com/dotnet/roslyn/issues/26505#issuecomment-386180135), I've executed code analysts in isolation, and the output is:

`error CS2001: Source file '/home/runner/work/appcyc-monorepo-demo/appcyc-monorepo-demo/identity-sts/obj/Release/net7.0/Appcyc.IdentitySts.GeneratedMSBuildEditorConfig.editorconfig' could not be found.`

The issue does not occur on my local machine, Ubunto (WSL2), nor in a docker build.

**Diagnostic Id**: MSB6006 & CS2001

**Expected Behavior**:

The build executes successfully.

**Actual Behavior**:

The build fails with `error MSB6006: "csc.dll" exited with code 139.`

I was getting the same `MSB6006: "csc.dll" exited with code 139` error in one of my public repos. I'm not sure if it's the same issue as @JSkimming is having, but figured I'd mention it in case it helped at all.

The ci workflow was running fine, but the codeql workflow was failing. The build continued to fail for a while, but the error changed from

> error MSB6006: "csc.dll" exited with code 139. ([link](https://github.com/xt0rted/tailwindcss-tag-helpers/actions/runs/5243840201/jobs/9469036943#step:4:162))
> Available sdks:
> 6.0.408 [/usr/share/dotnet/sdk]
> 7.0.105 [/usr/share/dotnet/sdk]
> 7.0.203 [/usr/share/dotnet/sdk]
> 7.0.302 [/usr/share/dotnet/sdk]

to

> error MSB6006: "csc.dll" exited with code 1. ([link](https://github.com/xt0rted/tailwindcss-tag-helpers/actions/runs/5243844955/jobs/9872439430#step:4:163))
> Available sdks:
> 6.0.411 [/usr/share/dotnet/sdk]
> 7.0.108 [/usr/share/dotnet/sdk]
> 7.0.203 [/usr/share/dotnet/sdk]
> 7.0.305 [/usr/share/dotnet/sdk]

It eventually went away without changing anything but I'm not sure why. The first successful build was when I bumped the pinned sdk from 7.0.302 to 7.0.304 and it hasn't been back since. You can see the failed runs [here](https://github.com/xt0rted/tailwindcss-tag-helpers/actions/workflows/codeql-analysis.yml).

I just restored the old branch that was giving the 139 error code and that [still fails](https://github.com/xt0rted/tailwindcss-tag-helpers/actions/runs/5243844955/jobs/9872439430?pr=235#step:4:163) but with an error code of 1 now. My assumption is even though it's pinned to 7.0.302 the available sdks on the vm are different, there's newer versions, and that's causing the difference.

The ubuntu 20.04 and 22.04 vms have different pre-installed sdk versions so that could be why a build world work on one but not the other, and why the issue could suddenly come or go as those versions are updated/replaced.

--- a/src/coreclr/vm/precode.cpp
+++ b/src/coreclr/vm/precode.cpp
@@ -662,7 +662,15 @@ BOOL StubPrecode::IsStubPrecodeByASM(PCODE addr)
             *(WORD*)(pInstr + 5) == *(WORD*)((BYTE*)StubPrecodeCode + 5) &&
             *(DWORD*)(pInstr + SYMBOL_VALUE(StubPrecodeCode_Target_Offset)) == (DWORD)(pInstr + GetStubCodePageSize() + offsetof(StubPrecodeData, Target));
 #else // TARGET_X86
-    return memcmp(pInstr, (void*)PCODEToPINSTR((PCODE)StubPrecodeCode), (BYTE*)StubPrecodeCode_End - (BYTE*)StubPrecodeCode) == 0;
+    BYTE *pTemplateInstr = (BYTE*)PCODEToPINSTR((PCODE)StubPrecodeCode);
+    BYTE *pTemplateInstrEnd = (BYTE*)PCODEToPINSTR((PCODE)StubPrecodeCode_End);
+    while ((pTemplateInstr < pTemplateInstrEnd) && (*pInstr == *pTemplateInstr))
+    {
+        pInstr++;
+        pTemplateInstr++;
+    }
+
+    return pTemplateInstr == pTemplateInstrEnd;
 #endif // TARGET_X86
 }
 
@@ -779,7 +787,15 @@ BOOL FixupPrecode::IsFixupPrecodeByASM(PCODE addr)
         *(WORD*)(pInstr + 11) == *(WORD*)((BYTE*)FixupPrecodeCode + 11) &&
         *(DWORD*)(pInstr + SYMBOL_VALUE(FixupPrecodeCode_PrecodeFixupThunk_Offset)) == (DWORD)(pInstr + GetStubCodePageSize() + offsetof(FixupPrecodeData, PrecodeFixupThunk));
 #else // TARGET_X86
-    return memcmp(pInstr, (void*)PCODEToPINSTR((PCODE)FixupPrecodeCode), (BYTE*)FixupPrecodeCode_End - (BYTE*)FixupPrecodeCode) == 0;
+    BYTE *pTemplateInstr = (BYTE*)PCODEToPINSTR((PCODE)FixupPrecodeCode);
+    BYTE *pTemplateInstrEnd = (BYTE*)PCODEToPINSTR((PCODE)FixupPrecodeCode_End);
+    while ((pTemplateInstr < pTemplateInstrEnd) && (*pInstr == *pTemplateInstr))
+    {
+        pInstr++;
+        pTemplateInstr++;
+    }
+
+    return pTemplateInstr == pTemplateInstrEnd;
 #endif // TARGET_X86
 }
 

